bench/wtperf/wtperf_truncate.c

```diff
@@ -145,11 +152,27 @@ run_truncate(CONFIG *cfg, CONFIG_THREAD *thread,
 	trunc_cfg->last_total_inserts = trunc_cfg->total_inserts;
 
 	/* We are done if there isn't enough data to trigger a new milestone. */
-	if (trunc_cfg->expected_total <= trunc_cfg->needed_stones)
+	if (trunc_cfg->expected_total <= thread->workload->truncate_count)
 		return (0);
 
+	/*
+	 * If we are falling behind and using more than one stone per lap we
+	 * should widen the stone gap for this lap to try and catch up quicker.
+	 */
+	if (trunc_cfg->expected_total >
+	    thread->workload->truncate_count + trunc_cfg->stone_gap) {
+		trunc_cfg->catchup_multiplier =
+		    WT_MIN(trunc_cfg->catchup_multiplier + 1,
```

Comment:
@daveh86 I figured out what these WT_MIN and WT_MAX statements do, but it wasn't obvious. Could you add some comments?